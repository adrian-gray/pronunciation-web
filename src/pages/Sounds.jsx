import React from 'react'

import {
  Paper,
  Typography,
  withStyles
} from '@material-ui/core'

import data from './../data/sounds'
import SEO from './../components/SEO'
import Phoneme from './../components/Phoneme'

const styles = (theme) => ({
  page: theme.page
})

const Sounds = (props) => {
  const { classes } = props

  const phonemes = data.phonemes
  const list = Object.keys(phonemes)
  const cards = list.map(phoneme => {
    const ipa = phonemes[phoneme]['ipa']
    const title = phonemes[phoneme]['title']
    const tag = phonemes[phoneme]['tag']
    const words = phonemes[phoneme]['common words']
    return <Phoneme key={phoneme} phoneme={phoneme} ipa={ipa} title={title} tag={tag} words={words} />
  })

  return (
    <Paper className={classes.page}>
      <SEO meta='sounds' />
      <Typography variant='display1' gutterBottom>
        {'Phoenetic English Pronunciation'}
      </Typography>
      {cards}
    </Paper>
  )
}

export default withStyles(styles)(Sounds)
