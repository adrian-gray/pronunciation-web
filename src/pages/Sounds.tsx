import React from "react";

import Container from "react-bootstrap/Container";

import data from "./../data/sounds.json";
import SEO from "../components/SEO";
import Phoneme from "../components/Phoneme";

export default props => {
  const phonemes = data.phonemes;
  const list = Object.keys(phonemes);
  const cards = list.map(phoneme => {
    const ipa = phonemes[phoneme]["ipa"];
    const title = phonemes[phoneme]["title"];
    const tag = phonemes[phoneme]["tag"];
    const words = phonemes[phoneme]["activities"]["common words"].words;
    return (
      <Phoneme
        key={phoneme}
        phoneme={phoneme}
        ipa={ipa}
        title={title}
        tag={tag}
        words={words}
      />
    );
  });

  return (
    <Container>
      <SEO meta="sounds" />
      <h4>{"English Pronunciation Sounds"}</h4>
      {cards}
    </Container>
  );
};
