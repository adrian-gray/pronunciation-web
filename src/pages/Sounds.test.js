import React from "react";
import { shallow } from "enzyme";
import { render, cleanup } from "@testing-library/react";

import Sounds from "./Sounds";

afterEach(cleanup);

test("Sounds", async () => {
  const props = {};

  const container = shallow(<Sounds {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
