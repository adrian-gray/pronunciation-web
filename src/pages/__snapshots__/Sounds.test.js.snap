// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sounds 1`] = `undefined`;
